### PR TITLE
Fix new version

### PR DESCRIPTION
**Pull request by shirser121**

### CODE REVIEW

**Review by LucianoHanna — APPROVED** [review]
Looks good to me

**Review by Rakambda — APPROVED** [review]

**Review by jtouris — APPROVED** [review]

**Review by tuyuribr — APPROVED** [review]

**Review by aliyss — APPROVED** [review]
lgtm

**Comment by jurajmatus** [comment]
`chat.preload` is a synchronous method which only spawns `loadEarlierMsgs` and returns immediately, which results in always having 1 message loaded from chat

I suggest to apply this patch in addition to the changes here: https://github.com/instea/whatsapp-web.js/commit/36882c33a6b785a36dc6a88e4a52889e78268c54

**Comment by XxQuickSilverZz** [comment]
> `chat.preload` is a synchronous method which only spawns `loadEarlierMsgs` and returns immediately, which results in always having 1 message loaded from chat
> 
> I suggest to apply this patch in addition to the changes here: [instea@36882c3](https://github.com/instea/whatsapp-web.js/commit/36882c33a6b785a36dc6a88e4a52889e78268c54)

i still get only the last message. Changed the lines above...

**Review by Yuri-Lima — APPROVED** [review]
Fine

File: src/Client.js [hunk 1]
```diff
@@ -809,8 +809,8 @@ class Client extends EventEmitter {
                 return true;
             }
             const MAX_PIN_COUNT = 3;
-            if (window.Store.Chat.models.length > MAX_PIN_COUNT) {
-                let maxPinned = window.Store.Chat.models[MAX_PIN_COUNT - 1].pin;
+            if (window.Store.Chat._models.length > MAX_PIN_COUNT) {
```
Comment (Yuri-Lima) [inline src/Client.js:812]: what about that?
`
window.Store.Chat.models.length ? window.Store.Chat.models.length : window.Store.Chat._models.length
`

File: src/Client.js [hunk 2]
```diff
@@ -1056,7 +1056,7 @@ class Client extends EventEmitter {
     async getChatsByLabelId(labelId) {
         const chatIds = await this.pupPage.evaluate(async (labelId) => {
             const label = window.Store.Label.get(labelId);
-            const labelItems = label.labelItemCollection.models;
+            const labelItems = label.labelItemCollection._models;
```
Comment (Yuri-Lima) [inline src/Client.js:1059]: Same here
`
const labelItems = label.labelItemCollection.models || label.labelItemCollection._models;
`

File: src/util/Injected.js [hunk 3]
```diff
@@ -416,7 +416,7 @@ exports.LoadUtils = () => {
     };
 
     window.WWebJS.getChats = async () => {
-        const chats = window.Store.Chat.models;
+        const chats = window.Store.Chat._models;
```
Comment (Yuri-Lima) [inline src/util/Injected.js:419]: Same
`
const chats = window.Store.Chat.models || window.Store.Chat._models;
`

File: src/util/Injected.js [hunk 4]
```diff
@@ -448,7 +448,7 @@ exports.LoadUtils = () => {
     };
 
     window.WWebJS.getContacts = () => {
-        const contacts = window.Store.Contact.models;
+        const contacts = window.Store.Contact._models;
```
Comment (Yuri-Lima) [inline src/util/Injected.js:451]: Same
`
const contacts = window.Store.Contact.models || window.Store.Contact._models;
`

File: src/util/Injected.js [hunk 5]
```diff
@@ -541,7 +541,7 @@ exports.LoadUtils = () => {
     };
 
     window.WWebJS.getLabels = () => {
-        const labels = window.Store.Label.models;
+        const labels = window.Store.Label._models;
```
Comment (Yuri-Lima) [inline src/util/Injected.js:544]: Same
`
const labels = window.Store.Label.models || window.Store.Label._models;
`

**Review by Yuri-Lima — CHANGES_REQUESTED** [review]

**Comment by LucianoHanna** [comment]
> 



> after apply this patch, i got this error
> 
> ```
> Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'getContact')
>     at __puppeteer_evaluation_script__:2:34
>     at ExecutionContext._evaluateInternal (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
>     at processTicksAndRejections (internal/process/task_queues.js:97:5)
>     at async ExecutionContext.evaluate (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
>     at async Client.getContactById (/src/node_modules/whatsapp-web.js/src/Client.js:671:23)
>     at async /src/app.js:303:27
> ```
> 
> i think the right method is `getContacts`. or am i miss something ?

`return window.WWebJS.getContact(contactId);`

Error is probably at this line, I think it was because window.WWebJS is undefined.

Maybe there's another problem causing that. Others functions are not working in your app or just that function?

**Comment by abudawud** [comment]
> after apply this patch, i got this error
> 
> ```
> Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'getContact')
>     at __puppeteer_evaluation_script__:2:34
>     at ExecutionContext._evaluateInternal (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
>     at processTicksAndRejections (internal/process/task_queues.js:97:5)
>     at async ExecutionContext.evaluate (/src/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
>     at async Client.getContactById (/src/node_modules/whatsapp-web.js/src/Client.js:671:23)
>     at async /src/app.js:303:27
> ```
> 
> i think the right method is `getContacts`. or am i miss something ?

so sorry, this error just happened in my server. i test it on my local development it's working well

**Review by pedroslopez — APPROVED** [review]
Made a few changes to match whatsapp's internal usage. Thanks for the fix!

**Pull request merged**